src/databricks/labs/ucx/hive_metastore/objects.py

```diff
@@ -0,0 +1,36 @@
+from dataclasses import dataclass
```

Comment:
@nfx: Have a look at this module.

Notes:
- Needed to be a separate module to avoid circular imports
- Maybe we want to introduce a parent object [securable object](https://docs.databricks.com/en/sql/language-manual/sql-ref-privileges.html#securable-objects) (later), to have a common interface for securable objects
- Note some inconsistencies across this code base and including the sdk:
  - `SchemaInfo(catalog_name="catalog")` vs `Schema(catalog="catalog")`. I choose the later to be consistent with the already existing `Table` class
  - Should we have a `Schema` or `Database`?
  - When do we use `full_name` and when `key`?
  - More?

Reply:
> Needed to be a separate module to avoid circular imports

Not a problem anymore after using a Protocol, https://github.com/databrickslabs/ucx/pull/2955#discussion_r1801182440

Reply:
Added the above as code comments